### PR TITLE
lf_interop_qos.py: Add monitor_cx to wait for CX creation

### PR DESCRIPTION
**Pull request by kadalinikhita**
- Waits up to 20 retries for all CXs (connections) to be created
- Devices will be excluded from the test if their cross-connections are not created.

### CODE REVIEW

File: py-scripts/lf_interop_qos.py [hunk 1]
```diff
@@ -581,6 +582,66 @@ def create_cx(self):
                 count += 1
             logger.info("cross connections with TOS type created.")
 
+    def monitor_cx(self):
+        """
+        This function waits for upto 20 iterations to allow all CXs (connections) to be created.
```
Comment (a-gavin) [inline py-scripts/lf_interop_qos.py:587]: nit type in `upto`

File: py-scripts/lf_interop_qos.py [hunk 2]
```diff
@@ -581,6 +582,66 @@ def create_cx(self):
                 count += 1
             logger.info("cross connections with TOS type created.")
 
+    def monitor_cx(self):
```
Comment (a-gavin) [inline py-scripts/lf_interop_qos.py:585]: Is there a reason the `monitor()` function in the L3 CXProfile code is not suitable? Can understand if logic here needs to be more test-specific, but want to make sure we don't duplicate code if functionality is already there.

**Review by a-gavin — CHANGES_REQUESTED** [review]

**Comment by a-gavin** [comment]
I see that force push was to grab latest upstream patches. Will await feedback on comments

File: py-scripts/lf_interop_qos.py (hunk 2 above)
Comment (kadalinikhita) [reply py-scripts/lf_interop_qos.py:572]: The monitor() function from the L3 CXProfile cannot be directly reused for the QoS test because the QoS implementation already includes test-specific logic within its own monitor() method.

The change I’ve introduced adds a new monitor_cx() function, which runs before starting and monitoring the connections. This function verifies whether each connection (CX) has been successfully created, retrying up to 20 times. If a connection is still not created after these attempts, the corresponding device is removed from the test.
